### DISCUSSION TITLE
Add unit conversion support to the number entity

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#771] [Entity Models] [answered]
Proposal:
Add unit conversion support to the number entity. The unit conversion will initially be implemented for temperatures, with the option to extend it further in the future.

Background:
Similar to numerical sensors, number entities may represent a physical quantity where the preferred display unit depends on the user's locale.

Implementation suggestion:
For the same reasons as for `sensor`, it's preferred to implement the unit conversion in the base number entity instead of in each integration which has number entities.

**Comment by balloob** [comment]
The challenge here is that it not only impacts the output (state) but also what is sent to the entity (service). The service descriptions are currently static.

**Reply by emontnemery** [reply]
This is not an issue because the `set` service description does not include the entity's unit of measurement.

**Comment by emontnemery** [comment, marked as answer]
Proposed answer:
Add unit conversion support to the number entity as suggested in the architecture issue

Implementation proposal in https://github.com/home-assistant/core/pull/73233
